internal/app/machined/pkg/controllers/k8s/apiserver_loadbalancer_config.go

```diff
@@ -0,0 +1,154 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package k8s
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/cosi-project/runtime/pkg/controller"
+	"github.com/cosi-project/runtime/pkg/safe"
+	"github.com/cosi-project/runtime/pkg/state"
+	"github.com/siderolabs/gen/channel"
+	"github.com/siderolabs/go-pointer"
+	"go.uber.org/zap"
+
+	"github.com/siderolabs/talos/pkg/machinery/resources/config"
+	"github.com/siderolabs/talos/pkg/machinery/resources/k8s"
+)
+
+// APILoadBalancerConfigController creates config for load balancer.
+type APILoadBalancerConfigController struct{}
```

Comment:
I don't think so, because `transform.Controller` relies on `touched` map, and we explicitly do not want to preserve the resulting resource if we didn't have both parts.

Also I'm pretty sure `transform.Controller` is tailored towards collection of resources and disallowing deletion until child resource is removed.

Reply:
gotcha :heavy_check_mark: